<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1278</width>
    <height>573</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Robotics Project1</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="lb_task1">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>18</pointsize>
      <weight>75</weight>
      <italic>false</italic>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Task1</string>
    </property>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>90</y>
      <width>311</width>
      <height>231</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="0">
      <widget class="QLineEdit" name="le_nx">
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="placeholderText">
        <string>nx</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLineEdit" name="le_ny">
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="placeholderText">
        <string>ny</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLineEdit" name="le_nz">
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="placeholderText">
        <string>nz</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLineEdit" name="le_nw">
       <property name="text">
        <string>0</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="le_ox">
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="placeholderText">
        <string>ox</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLineEdit" name="le_oy">
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="placeholderText">
        <string>oy</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QLineEdit" name="le_oz">
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="placeholderText">
        <string>oz</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QLineEdit" name="le_ow">
       <property name="text">
        <string>0</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QLineEdit" name="le_ax">
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="placeholderText">
        <string>ax</string>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QLineEdit" name="le_ay">
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="placeholderText">
        <string>ay</string>
       </property>
      </widget>
     </item>
     <item row="2" column="2">
      <widget class="QLineEdit" name="le_az">
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="placeholderText">
        <string>az</string>
       </property>
      </widget>
     </item>
     <item row="3" column="2">
      <widget class="QLineEdit" name="le_aw">
       <property name="text">
        <string>0</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="0" column="3">
      <widget class="QLineEdit" name="le_px">
       <property name="inputMask">
        <string/>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="placeholderText">
        <string>px</string>
       </property>
      </widget>
     </item>
     <item row="1" column="3">
      <widget class="QLineEdit" name="le_py">
       <property name="inputMask">
        <string/>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="placeholderText">
        <string>py</string>
       </property>
      </widget>
     </item>
     <item row="2" column="3">
      <widget class="QLineEdit" name="le_pz">
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="placeholderText">
        <string>pz</string>
       </property>
      </widget>
     </item>
     <item row="3" column="3">
      <widget class="QLineEdit" name="le_pw">
       <property name="text">
        <string>1</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="lb_noap">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>40</y>
      <width>231</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
      <weight>75</weight>
      <italic>false</italic>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Cartesian Point (n, o, a, p)</string>
    </property>
   </widget>
   <widget class="QLabel" name="lb_task2">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>360</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>18</pointsize>
      <weight>75</weight>
      <italic>false</italic>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Task2</string>
    </property>
   </widget>
   <widget class="QLabel" name="lb_thetas">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>380</y>
      <width>391</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
      <weight>75</weight>
      <italic>false</italic>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Joint Variables (theta 1 to 6) (unit: degree)</string>
    </property>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>420</y>
      <width>531</width>
      <height>80</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLineEdit" name="le_th1">
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="placeholderText">
        <string>theta1</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="le_th2">
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="placeholderText">
        <string>theta2</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="le_th3">
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="placeholderText">
        <string>theta3</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="le_th4">
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="placeholderText">
        <string>theta4</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="le_th5">
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="placeholderText">
        <string>theta5</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="le_th6">
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="placeholderText">
        <string>theta6</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>340</y>
      <width>521</width>
      <height>16</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_dotask1">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>230</y>
      <width>81</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Compute</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_dotask2">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>510</y>
      <width>81</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Compute</string>
    </property>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>330</y>
      <width>281</width>
      <height>181</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="1" column="1">
      <widget class="QLabel" name="lb_oy">
       <property name="text">
        <string>oy</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="lb_ny">
       <property name="text">
        <string>ny</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="lb_nw">
       <property name="text">
        <string>0</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QLabel" name="lb_ax">
       <property name="text">
        <string>ax</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="lb_nz">
       <property name="text">
        <string>nz</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="3" column="2">
      <widget class="QLabel" name="lb_aw">
       <property name="text">
        <string>0</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QLabel" name="lb_ay">
       <property name="text">
        <string>ay</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="lb_nx">
       <property name="text">
        <string>nx</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QLabel" name="lb_ow">
       <property name="text">
        <string>0</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLabel" name="lb_ox">
       <property name="text">
        <string>ox</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QLabel" name="lb_oz">
       <property name="text">
        <string>oz</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="2" column="2">
      <widget class="QLabel" name="lb_az">
       <property name="text">
        <string>az</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="0" column="3">
      <widget class="QLabel" name="lb_px">
       <property name="text">
        <string>px</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="1" column="3">
      <widget class="QLabel" name="lb_py">
       <property name="text">
        <string>py</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="2" column="3">
      <widget class="QLabel" name="lb_pz">
       <property name="text">
        <string>pz</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="3" column="3">
      <widget class="QLabel" name="lb_pw">
       <property name="text">
        <string>1</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="Line" name="line_2">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>20</y>
      <width>20</width>
      <height>531</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="QLabel" name="lb_result">
    <property name="geometry">
     <rect>
      <x>590</x>
      <y>20</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>18</pointsize>
      <weight>75</weight>
      <italic>false</italic>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Result</string>
    </property>
   </widget>
   <widget class="QLabel" name="lb_variables">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>60</y>
      <width>421</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
      <weight>75</weight>
      <italic>false</italic>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Variables </string>
    </property>
   </widget>
   <widget class="QLabel" name="lb_noap_2">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>270</y>
      <width>231</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
      <weight>75</weight>
      <italic>false</italic>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Cartesian Point (n, o, a, p)</string>
    </property>
   </widget>
   <widget class="QTableView" name="tableView">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>110</y>
      <width>651</width>
      <height>151</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_task1sample1">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>130</y>
      <width>81</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Sample1</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_task1sample2">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>180</y>
      <width>81</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Sample2</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_task2sample1">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>510</y>
      <width>81</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Sample1</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_task2sample1_2">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>510</y>
      <width>81</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Sample1_2</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_task2sample2">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>510</y>
      <width>81</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Sample2</string>
    </property>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>btn_dotask1</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>Task1()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>500</x>
     <y>244</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>156</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_dotask2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>Task2()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>450</x>
     <y>520</y>
    </hint>
    <hint type="destinationlabel">
     <x>520</x>
     <y>474</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_task2sample2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>Task2Sample2()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>329</x>
     <y>530</y>
    </hint>
    <hint type="destinationlabel">
     <x>327</x>
     <y>558</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_task2sample1_2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>Task2Sample1_2()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>228</x>
     <y>526</y>
    </hint>
    <hint type="destinationlabel">
     <x>227</x>
     <y>547</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_task2sample1</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>Task2Sample1()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>127</x>
     <y>529</y>
    </hint>
    <hint type="destinationlabel">
     <x>128</x>
     <y>493</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_task1sample1</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>Task1Sample1()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>463</x>
     <y>149</y>
    </hint>
    <hint type="destinationlabel">
     <x>468</x>
     <y>103</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_task1sample2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>Task1Sample2()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>480</x>
     <y>196</y>
    </hint>
    <hint type="destinationlabel">
     <x>421</x>
     <y>200</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>Task2()</slot>
  <slot>Task1()</slot>
  <slot>Task1Sample1()</slot>
  <slot>Task2Sample2()</slot>
  <slot>Task1Sample2()</slot>
  <slot>Task2Sample1()</slot>
  <slot>Task2Sample1_2()</slot>
 </slots>
</ui>
